operator/pkg/ciliumendpointslice/cescache.go

```diff
@@ -98,3 +170,54 @@ func (c *CESCache) getEncryptionKey(nodeName NodeName) (EncryptionKey, bool) {
 	}
 	return 0, false
 }
+
+// Return if given CID is present in cache
+func (c *CESCache) hasCID(cid CID, gidLabels Label) bool {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+	_, ok := c.cidToGidLabels[cid]
+	return ok
+}
+
+// Return the selected CID for the given GID labels
+func (c *CESCache) GetSelectedId(gid Label) (CID, bool) {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+
+	if gidData, ok := c.globalIdLabelsToCIDSet[gid]; ok {
+		if gidData.selectedID != "" {
+			return gidData.selectedID, true
+		}
+	}
+	return "", false
+}
+
+// Clean up globalIdLabelsToCIDSet map, if no label state exists.
+// Caller should hold cache lock.
+func (c *CESCache) cleanLabelsMap(gidLabels Label) {
+	if secId, ok := c.globalIdLabelsToCIDSet[gidLabels]; ok {
+		if secId.ceps.Len() == 0 && secId.ids.Len() == 0 {
+			delete(c.globalIdLabelsToCIDSet, gidLabels)
+		}
+	}
+}
+
+func (cid CID) String() string {
+	return string(cid)
+}
+
+// SetSelectedID will update the selectedID to the given CID if not set.
+// If findNextCID is true, it will find the next available CID
+// and update selectedID to it, or set to empty if no CIDs are available.
+// Caller should hold cache lock.
+func (s *SecIDs) setSelectedID(newCID CID, findNextCID bool) {
```

Comment:
Is there any case in which `newCID` is empty, and `findNextCID` is false? Otherwise the latter could be dropped for simplicity.